robotbase/src/main/java/com/team2357/frc2022/commands/ExampleTrajectoryCommand.java

```diff
@@ -0,0 +1,80 @@
+package com.team2357.frc2022.commands;
+
+import java.util.List;
+
+import com.team2357.frc2022.Constants;
+import com.team2357.lib.commands.CommandLoggerBase;
+import com.team2357.lib.subsystems.drive.FalconTrajectoryDriveSubsystem;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+
+public class ExampleTrajectoryCommand extends CommandLoggerBase{
+    FalconTrajectoryDriveSubsystem m_driveSub;
+
+    public ExampleTrajectoryCommand(FalconTrajectoryDriveSubsystem driveSub) {
+        m_driveSub = driveSub;
+    }
+
+    public Command getRamsete() {
+        var autoVoltageConstraint = new DifferentialDriveVoltageConstraint(
+            new SimpleMotorFeedforward(
+                Constants.DRIVE.KS_VOLTS,
+                Constants.DRIVE.KV_VOLTS_SECONDS_PER_METER,
+                Constants.DRIVE.KA_VOLTS_SECONDS_SQUARED_PER_METER),
+            Constants.DRIVE.DRIVE_KINEMATICS,
+            Constants.DRIVE.MAX_VOLTAGE);
+
+        // Create config for trajectory
+        TrajectoryConfig config = new TrajectoryConfig(
+            Constants.DRIVE.MAX_SPEED_METERS_PER_SECOND,
+            Constants.DRIVE.MAX_ACCELERATION_METERS_PER_SECOND_SQUARED)
+                // Add kinematics to ensure max speed is actually obeyed
+                .setKinematics(Constants.DRIVE.DRIVE_KINEMATICS)
+                // Apply the voltage constraint
+                .addConstraint(autoVoltageConstraint);
+
+        // An example trajectory to follow. All units in meters.
+        Trajectory exampleTrajectory = TrajectoryGenerator.generateTrajectory(
```

Comment:
I'm thinking we probably want to separate the creation of the trajectory poses here, the rest of the above code can be the same for all commands, and the bottom part as well. Maybe this line itself can be a `calculateTrajectory()` function that can be overridden by subclasses? That way we can keep all the boilerplate code above and below this in one place.

Reply:
Okay, I am moving this command out to a testing branch for now with my other trajectory things I was playing with.